Library/DiscUtils.Vhdx/LogEntry.cs

```diff
@@ -101,90 +101,84 @@ public static bool TryRead(Stream logStream, out LogEntry entry)
     {
         var position = logStream.Position;
 
-        var sectorBuffer = ArrayPool<byte>.Shared.Rent(LogSectorSize);
+        var bytesToRead = LogEntryHeader.ByteCount;
+        Span<byte> headerBuffer = stackalloc byte[bytesToRead];
+        if (logStream.ReadMaximum(headerBuffer) != bytesToRead)
+        {
+            entry = null;
+            return false;
+        }
+
+        var header = new LogEntryHeader();
+        header.ReadFrom(headerBuffer);
+
+        if (!header.IsValid)
+        {
+            entry = null;
+            return false;
+        }
+
+        var entryLength = checked((int)header.EntryLength);
+        var logEntryBuffer = ArrayPool<byte>.Shared.Rent(entryLength);
+
         try
         {
-            if (logStream.ReadMaximum(sectorBuffer, 0, LogSectorSize) != LogSectorSize)
-            {
-                entry = null;
-                return false;
-            }
+            headerBuffer.CopyTo(logEntryBuffer);
 
-            var sig = EndianUtilities.ToUInt32LittleEndian(sectorBuffer, 0);
-            if (sig != LogEntryHeader.LogEntrySignature)
+            bytesToRead = entryLength - LogEntryHeader.ByteCount;
+            if (logStream.ReadMaximum(logEntryBuffer, LogEntryHeader.ByteCount, bytesToRead) != bytesToRead)
             {
                 entry = null;
                 return false;
             }
 
-            var header = new LogEntryHeader();
-            header.ReadFrom(sectorBuffer.AsSpan(0, LogSectorSize));
-
-            if (!header.IsValid || header.EntryLength > logStream.Length)
```

Comment:
This data availability check would be broken when working within a stream starting at an offset > 0. ReadMaximum() below does the availability check correctly.